agent/api/task.go

```diff
@@ -108,6 +108,10 @@ type Task struct {
 	StartSequenceNumber int64
 	StopSequenceNumber  int64
 
+	// executionCredentialsID is the id of credentials used by agent to
+	// perform some action in the task level, like pull image from ecr
+	executionCredentialsID string
```

Comment:
Why isn't this persisted to the disk? How do we recover it on Agent restart?

Reply:
There is a trade off here, if we save this to the state file, then we don't need to read this field of the received acs message for this task. But if we don't save this to the state file, we will need to read that field of the task. Let me know your opinion.

Reply:
I think we *have to* persist this. Otherwise, how would we handle requests for containers that have already been started if the Agent restarts after the task has been transitioned to `RUNNING`?